aspnetcore/mvc/advanced/custom-model-binding.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Custom Model Binding | Microsoft Docs
+author: ardalis
+description: Customizing model binding in ASP.NET Core MVC.
+keywords: ASP.NET Core, model binding, custom model binder
+ms.author: riande
+manager: wpickett
+ms.date: 4/10/2017
+ms.topic: article
+ms.assetid: ebd98159-a028-4a94-b06c-43981c79c6be
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: mvc/advanced/custom-model-binding
+---
+# Custom Model Binding
+
+By [Steve Smith](http://ardalis.com)
+
+Model binding allows controller actions to work directly with model types, passed in as method arguments, rather than HTTP requests. Repetitive mapping between incoming request data and application models is consolidated into model binders. Developers can extend the built-in model binding functionality by implementing custom model binders.
+
+[View or download sample from GitHub](https://github.com/aspnet/Docs/)
+
+## The problem
+
+The default model binding supports simple types, complex types, collections, and more. The built in model binders should support the vast majority of developers' needs. However, the default model binders expect to bind text-based input from the request directly to model types. You might need to transform the input prior to binding it. One example of this is if you have input that represents a key that can be used to look up model data from a data store. You can use a custom model binder to fetch data based on the key and pass the result as the action's argument.
```

Comment:
I would mention that we have support for most of the common **data** types in the framework.